hack/update-featuregates.sh

```diff
@@ -28,3 +28,5 @@ source "${KUBE_ROOT}/hack/lib/init.sh"
 cd "${KUBE_ROOT}"
 
 go run test/featuregates_linter/main.go feature-gates update
+sed -i '1i\# This is a generated file. Do not edit directly.\n# Run hack/update-featuregates.sh to update it.' test/featuregates_linter/test_data/versioned_feature_list.yaml
```

Comment:
Shouldn't this be emitted by `go run test/featuregates_linter/main.go feature-gates update`, so we can have verify confirm that these lines exist?

https://github.com/kubernetes/kubernetes/blob/e9cde03b915e8697d64d2908efae536dd339fc73/hack/verify-featuregates.sh#L31

Right now we won't fail presubmit if someone edits these lines.

Reply:
Let me understand what you mean, you mean to put the text lines into the update and verify logic as well? Even though it has nothing to do with featuregate.

How is the way people modify text in hack/update-featuregates.sh different from how they modify text in code?


The reason I didn't put the logic into the code is because I didn't want to make the code more complicated, but I'd be happy to do that if necessary.
https://github.com/kubernetes/kubernetes/blob/99ff62e87a2d881f4839159d33519a49d6eee5d9/test/featuregates_linter/cmd/feature_gates.go#L162-L166

Reply:
> How is the way people modify text in hack/update-featuregates.sh different from how they modify text in code?

One idea is let `go run test/featuregates_linter/main.go feature-gates verify` working for featuregate and text lines.